src/org/zaproxy/zap/extension/accessControl/ExtensionAccessControl.java

```diff
@@ -586,4 +591,30 @@ public void importContextData(Context ctx, Configuration config) throws Configur
 		}
 	}
 
-}
\ No newline at end of file
+	protected int getScanProgress(int contextId) {
+
+		AccessControlScannerThread scannerThread = threadManager.getScannerThread(contextId);
+
+		if (scannerThread.isRunning() || scannerThread.hasRun()) {
+			double progress = scannerThread.getScanProgress()*1.0/scannerThread.getScanMaximumProgress()*100;
+			return (int)Math.round(progress);
+		} else {
+			log.error("Access control scan not running for context: " + contextId);
+			throw new IllegalStateException("A scan is not running for context: " + contextId);
+		}
+	}
+
+
+	protected String getScanStatus(int contextId) {
```

Comment:
This function should properly use curly brackets `{}` for the conditionals as elsewhere in the code (for consistency and readability).